### PR TITLE
Remove UI elements deprecated since 3.9

### PR DESCRIPTION
**Pull request by haticekaratay**
### Description

This pull request removes deprecated logic related to the "SAVE as FITS" option from the Collapse, Moment Maps, and Spectral Extraction plugins. These options were already deprecated and have now been fully removed from the UI. Instead, the process of exporting files is centralized in a separate Export Plugin, reducing redundancy.

To illustrate the updates, check the before-and-after videos that highlight the UI changes.

BEFORE:

https://github.com/user-attachments/assets/5050b13a-2e48-4d1f-a9ed-503bf6c392b9

AFTER:

https://github.com/user-attachments/assets/f1edd27a-941b-4200-bcd2-46715b407982

Fixes #<Issue Number>

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py [hunk 1]
```diff
@@ -672,50 +672,6 @@ def vue_spectral_extraction(self, *args, **kwargs):
     def vue_create_bg_spec(self, *args, **kwargs):
         self.extract_bg_spectrum(add_data=True)
 
-    def vue_save_as_fits(self, *args):
-        self._save_extracted_spec_to_fits()
-
-    def vue_overwrite_fits(self, *args):
-        """Attempt to force writing the spectral extraction if the user
-        confirms the desire to overwrite."""
-        self.overwrite_warn = False
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:681]: can this traitlet be removed too? 🤞

File: CHANGES.rst [hunk 2]
```diff
@@ -73,6 +73,9 @@ Cubeviz
 
 - Add missing styling to API hints entry for aperture_method in the spectral extraction plugin. [#3231]
 
+- Remove the ``save as fits`` option from the Collapse, Moment Maps and Spectral Extraction plugin; use the Export plugin instead. [#3256]
```
Comment (pllim) [inline CHANGES.rst:76]:
```suggestion
- Removed the deprecated ``save as fits`` option from the Collapse, Moment Maps, and Spectral Extraction plugins; use the Export plugin instead. [#3256]
```

**Review by pllim — COMMENTED** [review]
I like `+18 −409`. Just a minor comment on change log entry. Thanks!

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by pllim** [comment]
I think we agreed to put this in 4.1 and not 4.0.1 right, @kecnry ?

**Pull request merged**